services/webnn/ort/graph_builder_ort.cc

```diff
@@ -558,6 +607,103 @@ void GraphBuilderOrt::AddGemmOperation(const mojom::Gemm& gemm) {
                          attributes);
 }
 
+void GraphBuilderOrt::AddInstanceNormalizationOperation(
+    const mojom::InstanceNormalization& instance_normalization) {
+  const OperandDataType input_data_type =
+      GetOperand(instance_normalization.output_operand_id)
+          .descriptor.data_type();
+
+  const std::string input_name =
+      GetOperandName(instance_normalization.input_operand_id);
+  std::vector<const char*> input_names = {input_name.c_str()};
+
+  const std::vector<uint32_t>& input_shape =
+      GetOperand(instance_normalization.input_operand_id).descriptor.shape();
+  // TODO(crbug.com/387312212): Support NHWC layout
+  CHECK_EQ(context_properties_.input_operand_layout, InputOperandLayout::kNchw);
+  uint32_t input_channel = input_shape[1];
+  std::vector<uint32_t> constant_dims = {input_channel};
+
+  std::string scale_name, bias_name;
+  // ONNX requires scale and bias inputs.
+  if (instance_normalization.scale_operand_id) {
+    scale_name =
+        GetOperandName(instance_normalization.scale_operand_id.value());
+    input_names.push_back(scale_name.c_str());
+  } else {
+    std::vector<float> scale_data(input_channel, 1.0f);
+    switch (input_data_type) {
+      case OperandDataType::kFloat16: {
+        std::vector<uint16_t> scale_data_fp16 =
+            ConvertFloat32ToFloat16(scale_data);
+        scale_name = CreateInitializerAsRawData(
```

Comment:
Add a comment explaining why we need create raw data?

Reply:
> Add a comment explaining why we need create raw data?

Added. According to runtime error message, we can't use external data here.